tests/Nelmio/Alice/Fixtures/Parser/Methods/PhpTest.php

```diff
@@ -110,4 +110,31 @@ public function testIncludeFiles()
         ];
         $this->assertEquals($expectedData, $data);
     }
+
+    public function testIncludeFilesOrder()
+    {
+        $data = $this->parser->parse(__DIR__.'/../../../support/fixtures/include_order.php');
+
+        $expectedData = [
+            'Foo' =>
+                [
+                    'foo' =>
+                        [
+                            'id' => 1,
+                            'text' => '<word()>',
+                        ],
+                ],
+            'Bar' =>
+                [
+                    'bar' =>
+                        [
+                            'id' => 1,
+                            'text' => '<@foo->text>'
+                        ],
+                ],
+        ];
+
+        $this->assertEquals($expectedData, $data);
+        $this->assertTrue($expectedData === $data);
```

Comment:
You are right, but in this example you are comparing [0=>1,1=>2] with [0=>2,1=>1], hence the == operator return false because $first[0] != $second[0].

Instead, if you compare [1=>1,2=>2] and [2=>2,1=>1] the == operator return true because $first[1] == $second[1] and $first[2] == $second[2] no matter in what order they are.

On the other hand, if you use === operator you'll get false, because you are asking to compare also the orders.

[Here](http://php.net/manual/en/language.operators.array.php) you can find the documentation of array operators.